Classes_Homework_5/fromjson.py

```diff
@@ -0,0 +1,101 @@
+import argparse
+from jsonschema import validate, ValidationError, SchemaError
+from json import load, JSONDecodeError
+
+from modules.combine import Combine
+from modules.dates import Dates
+from modules.file import Files
+from modules.input import DateInput
+from modules.exceptions import PastDate, NoValue
+from exec_utils.configloader import Config
+from fromfile import WriteFromFile
+
+cnf = Config()
+f = Files()
+dte = Dates()
+di = DateInput(cnf.get_values("ERRORS", "past_date"),
+               cnf.get_values("PATTERNS", "date_format"),
+               dte.get_current_date())
+
+label1, label2, label3 = cnf.get_values("LABELS", "news_label"), \
+                         cnf.get_values("LABELS", "ad_label"), \
+                         cnf.get_values("LABELS", "recipe_label")
+decor, decor_length = cnf.get_values("RESTRICTIONS", "n"), \
+                      cnf.get_values("RESTRICTIONS", "count_n")
+city, txt, date, cal = cnf.get_values("PROPERTIES", "city"), \
+                       cnf.get_values("PROPERTIES", "text"), \
+                       cnf.get_values("PROPERTIES", "date"), \
+                       cnf.get_values("PROPERTIES", "kcal")
+
+
+class FromJson(WriteFromFile):  # inherit methods from WriteFromFile in fromfile.py.
+    def parse(self):
+        """Get path to json from command line, take from configs if not provided"""
+        parser = argparse.ArgumentParser(description="Process source file.")
+        parser.add_argument("path", type=str, nargs="?",
+                            default=self.def_source,
+                            help="full path to file.")
+        args = parser.parse_args()
+        return args.path
+
+    def get_schema(self):
```

Comment:
It'd be better to add `@classmethod` decorator here:

https://docs.quantifiedcode.com/python-anti-patterns/correctness/method_could_be_a_function.html#class-method-is-not-preceded-by-classmethod-decorator